walletnode/services/artworkregister/state/state_test.go

```diff
@@ -0,0 +1,234 @@
+			exptectedType:     StatusTaskCompleted,
+			expectedIsFinal:   true,
+		},
+	}
+
+	t.Run("group", func(t *testing.T) {
+		t.Parallel()
+
+		states := newTestStates().All()
+
+		for _, testCase := range testCases {
+
+			testCase := testCase
+
+			t.Run(fmt.Sprintf("created:%v/type:%v/final:%t", testCase.expectedCreatedAt, testCase.exptectedType, testCase.expectedIsFinal), func(t *testing.T) {
+				t.Parallel()
+
+				state := states[testCase.key]
+
+				assert.Equal(t, testCase.expectedCreatedAt, state.CreatedAt)
+				assert.Equal(t, testCase.exptectedType, state.Type)
+				assert.Equal(t, testCase.expectedIsFinal, state.isFinal)
+			})
+
+		}
+
+	})
+
+}
+
+func TestStatesLatest(t *testing.T) {
+	t.Parallel()
+
+	timeList := createTestTimeList()
+
+	testCases := []struct {
+		state  *State
+		status *Status
+	}{
+		{
+			state: newTestStates(),
+			status: &Status{
+				CreatedAt: timeList[2],
+				Type:      StatusTaskCompleted,
+				isFinal:   true,
+			},
+		}, {
+			state:  &State{},
+			status: nil,
+		},
+	}
+
+	t.Run("group", func(t *testing.T) {
+		t.Parallel()
+
+		for _, testCase := range testCases {
+			testCase := testCase
+
+			t.Run(fmt.Sprintf("status:%v", testCase.status), func(t *testing.T) {
+				t.Parallel()
+
+				latest := testCase.state.Latest()
+
+				assert.Equal(t, testCase.status, latest)
+			})
+		}
+
+	})
+
+}
+
+func TestStateSubscribe(t *testing.T) {
+	t.Parallel()
+
+	timeList := createTestTimeList()
+
+	testCases := []struct {
+		expectedCreatedAt time.Time
+		exptectedType     StatusType
+		expectedIsFinal   bool
+	}{
+		{
+			expectedCreatedAt: timeList[0],
+			exptectedType:     StatusTaskStarted,
+			expectedIsFinal:   false,
+		}, {
+			expectedCreatedAt: timeList[1],
+			exptectedType:     StatusConnected,
+			expectedIsFinal:   false,
+		}, {
+			expectedCreatedAt: timeList[2],
+			exptectedType:     StatusTaskCompleted,
+			expectedIsFinal:   true,
+		},
+	}
+
+	t.Run("group", func(t *testing.T) {
+		t.Parallel()
+
+		states := newTestStates()
+		sub, err := states.Subscribe()
+
+		assert.NoError(t, err)
+
+		for _, testCase := range testCases {
+			testCase := testCase
+
+			//no need run in parallel as only check state value
+			t.Run(fmt.Sprintf("created:%v/type:%v/final/%t", testCase.expectedCreatedAt, testCase.exptectedType, testCase.expectedIsFinal), func(t *testing.T) {
+				state := <-sub.statusCh
+
+				assert.Equal(t, testCase.expectedCreatedAt, state.CreatedAt)
+				assert.Equal(t, testCase.exptectedType, state.Type)
+				assert.Equal(t, testCase.expectedIsFinal, state.isFinal)
+			})
+		}
+	})
+
+}
+
+func TestStateUpdate(t *testing.T) {
+	t.Parallel()
+
+	t1, t2 := time.Now(), time.Now().Add(time.Second*10)
+
+	testCases := []struct {
+		createdAt  time.Time
+		typeStatus StatusType
+		isFinal    bool
+	}{
+		{
+			createdAt:  t1,
+			typeStatus: StatusTicketRegistered,
+			isFinal:    false,
+		}, {
+			createdAt:  t2,
+			typeStatus: StatusTaskRejected,
+			isFinal:    true,
+		},
```

Comment:
```suggestion
		{
			createdAt:  time.Now(),
			typeStatus: StatusTicketRegistered,
			isFinal:    false,
		}, {
			createdAt:  time.Now().Add(time.Second*10),
			typeStatus: StatusTaskRejected,
			isFinal:    true,
		},
```
It is preferable to embed the get value inside `testCases`, exception is when it's a function with multiples lines.